studybuddies/src/app/services/base-urlinterceptor.service.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+import {HttpHandler, HttpRequest, HttpInterceptor} from '@angular/common/http';
+import {environment} from '../../environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class BaseURLInterceptorService implements HttpInterceptor {
+
+  intercept(req: HttpRequest<any>, next: HttpHandler) {
+    if (!req.url.match(/^http(s)?:\/\/(.*)$/)) {
+      const url = `${environment.baseURL}${req.url}`.replace(/([^:]\/)\/+/g, '$1');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/Team-Studybuddies/Studybuddies-Frontend/pullRequest?prid=7202842)
```suggestion
      const url = `${environment.baseURL}${req.url}`.replace(/([^:]\/)\/+/g, "$1");
```